### PR TITLE
ledger-udev-rules: 2021-09-10 -> 2024-02-12

### PR DESCRIPTION
**Pull request by ToasterUwU**
## Description of changes

This updates the udev rules that are needed for ledger live desktop to be able to connect to the ledger hardware. The current version of these udev rules no longer works. 

The Ledger Live Desktop App couldn't connect to my Ledger Nano X. 
I tried not using `hardware.ledger.enable` but instead adding the current rules from the official repo "manually" with `service.udev.extraRules` and that fixed the issue. Thats how i confirmed the outdated udev rules are the problem.

This is my first time contributing to Nix(OS), so im not sure i did everything right, but i certainly tried. If there are any issues, please guide me through them.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ToasterUwU** [comment]
Ok, i switched to my branch for my system to test if it works after reboot and all, and yes. This fixes the issue and there are no other issues caused by it as far as i can tell.

**Comment by ToasterUwU** [comment]
Im not sure if this needs backporting or not, so i will trust more experienced people with that decision, but i can confirm that on the current version of Ledger Live Desktop, the old udev rules are not working, so if the versions in stable are similar or the same, this should be backported as well.

**Comment by Scrumplex** [comment]
> Im not sure if this needs backporting or not, so i will trust more experienced people with that decision, but i can confirm that on the current version of Ledger Live Desktop, the old udev rules are not working, so if the versions in stable are similar or the same, this should be backported as well.

Looking at the diff of the upstream repo, it seems like they refactored the rules but it seems like all devices should still work just as before.
https://github.com/LedgerHQ/udev-rules/compare/2776324af6df36c2af4d2e8e92a1c98c281117c9..f474382e370c9fa2a2207e6e675b9b364441aed7

I'd say this is a good candidate for backporting :D

File: pkgs/os-specific/linux/ledger-udev-rules/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation {
   pname = "ledger-udev-rules";
-  version = "unstable-2021-09-10";
+  version = "unstable-2024-05-15";
```
Comment (Scrumplex) [inline pkgs/os-specific/linux/ledger-udev-rules/default.nix:5]: Unstable packages should either start with the version the src is based on or with 0 if there is no version.

```suggestion
  version = "0-unstable-2024-05-15";
```

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 311899` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ledger-udev-rules</li>
  </ul>
</details>

File: pkgs/os-specific/linux/ledger-udev-rules/default.nix [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation {
   pname = "ledger-udev-rules";
-  version = "unstable-2021-09-10";
+  version = "0-unstable-2024-05-15";
```
Comment (asymmetric) [inline pkgs/os-specific/linux/ledger-udev-rules/default.nix:5]: Why the `0-`?

Comment (asymmetric) [reply pkgs/os-specific/linux/ledger-udev-rules/default.nix:5]: Also, this should be the date of the upstream commit, so 
```suggestion
  version = "unstable-2024-02-12";
```

**Review by asymmetric — CHANGES_REQUESTED** [review]
Thanks for the update -- a small change.

File: pkgs/os-specific/linux/ledger-udev-rules/default.nix (hunk 2 above)
Comment (ToasterUwU) [reply pkgs/os-specific/linux/ledger-udev-rules/default.nix:5]: > Why the `0-`?

Thats a relatively new convention that Scrumplex mentioned in his review: https://github.com/NixOS/nixpkgs/pull/311899#discussion_r1601450578

Comment (Scrumplex) [reply pkgs/os-specific/linux/ledger-udev-rules/default.nix:5]: > Why the `0-`?

https://github.com/NixOS/nixpkgs/pull/234201

Comment (ToasterUwU) [reply pkgs/os-specific/linux/ledger-udev-rules/default.nix:5]: > Also, this should be the date of the upstream commit, so

Ohh makes sense, i will change that but leave the 0 like Scrumplex suggested.

**Comment by ToasterUwU** [comment]
Oops, got excited and forgot to double check.

Will do in a moment.

**Review by ToasterUwU — COMMENTED** [review]
Ok, corrected the commit name as well. Should hopefully be good now

**Review by asymmetric — APPROVED** [review]

**Pull request merged**